protocol-specs/generators/proof_of_misbehavior_transaction/index.js

```diff
@@ -0,0 +1,531 @@
+		input: {
+			reportingAccount: accounts.reporter,
+			targetAccount: accounts.forger,
+			networkIdentifier,
+			transaction: unsignedTransaction,
+		},
+		output: tx,
+	};
+};
+
+/*
+	Scenario 2:
+
+	b1.height>b2.maxHeightPreviouslyForged
+*/
+
+const scenario2Header1 = {
+	version: 1,
+	timestamp: 1,
+	previousBlockId: 1,
+	seedReveal: '1',
+	height: 200000,
+	maxHeightPreviouslyForged: 100000,
+	maxHeightPrevoted: 100000,
+	numberOfTransactions: 0,
+	totalAmount: 0,
+	totalFee: 10000000000,
+	reward: 10000000000,
+	payloadLength: 0,
+	payloadHash: hash(Buffer.alloc(0)).toString('hex'),
+	generatorPublicKey:
+		'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+};
+
+scenario2Header1.blockSignature = sign(
+	scenario2Header1,
+	forgerKeyPair.privateKeyBytes.toString('hex'),
+);
+
+const scenario2Header2 = {
+	version: 1,
+	timestamp: 1,
+	previousBlockId: 1,
+	seedReveal: '1',
+	height: 200000,
+	maxHeightPreviouslyForged: 100000,
+	maxHeightPrevoted: 50000,
+	numberOfTransactions: 0,
+	totalAmount: 0,
+	totalFee: 10000000000,
+	reward: 10000000000,
+	payloadLength: 0,
+	payloadHash: hash(Buffer.alloc(0)).toString('hex'),
+	generatorPublicKey:
+		'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+};
+
+scenario2Header2.blockSignature = sign(
+	scenario2Header2,
+	forgerKeyPair.privateKeyBytes.toString('hex'),
+);
+
+const generateValidProofOfMisbehaviorTransactionForScenario2 = () => {
+	const unsignedTransaction = {
+		senderPublicKey: accounts.reporter.publicKey,
+		nonce: '1',
+		fee: '1500000000',
+		type: 15,
+		asset: {
+			header1: scenario2Header1,
+			header2: scenario2Header2,
+		},
+		signatures: [],
+	};
+
+	const tx = {
+		...unsignedTransaction,
+		asset: { ...unsignedTransaction.asset },
+		signatures: [],
+	};
+
+	const signBytes = serialize(tx);
+
+	tx.signatures.push(
+		createSignatureObject(signBytes, accounts.reporter).signature,
+	);
+
+	const id = getId(
+		Buffer.concat([
+			signBytes,
+			Buffer.from('01', 'hex'),
+			Buffer.from(tx.signatures[0], 'hex'),
+		]),
+	);
+
+	tx.id = id;
+
+	return {
+		input: {
+			reportingAccount: accounts.reporter,
+			targetAccount: accounts.forger,
+			networkIdentifier,
+			transaction: unsignedTransaction,
+		},
+		output: tx,
+	};
+};
+
+/*
+	Scenario 3:
+
+	b1.maxHeightPrevoted>b2.maxHeightPrevoted
+*/
+
+const scenario3Header1 = {
+	version: 1,
+	timestamp: 1,
+	previousBlockId: 1,
+	seedReveal: '1',
+	height: 200000,
+	maxHeightPreviouslyForged: 100000,
+	maxHeightPrevoted: 100001,
+	numberOfTransactions: 0,
+	totalAmount: 0,
+	totalFee: 10000000000,
+	reward: 10000000000,
+	payloadLength: 0,
+	payloadHash: hash(Buffer.alloc(0)).toString('hex'),
+	generatorPublicKey:
+		'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+};
+
+scenario3Header1.blockSignature = sign(
+	scenario3Header1,
+	forgerKeyPair.privateKeyBytes.toString('hex'),
+);
+
+const scenario3Header2 = {
+	version: 1,
+	timestamp: 1,
+	previousBlockId: 1,
```

Comment:
`previousBlockId` length should be around `10620616195853047363`, and it should be string

Reply:
what is `10620616195853047363`?

Reply:
does it matter what the values are for these properties and below? only the BFT related ones will be tested.

Reply:
i think type should be correct, otherwise getBytes wont work properly (or at least in the same way), and it's better to use some realistic value IMO
`10620616195853047363` this was some random block id taken from the test